### PR TITLE
Bump deps and project default versions

### PR DESCRIPTION
**Pull request by lilic**
This is in preparation of the release that should happen today. 

Seems like the `IPFamily` field was [removed](https://kubernetes.io/docs/setup/release/notes/#urgent-upgrade-notes) from the Service object. 

**Release Note Template (will be copied)**

```release-note:REPLACEME

```

### CODE REVIEW

File: pkg/k8sutil/k8sutil.go [hunk 1]
```diff
@@ -118,7 +118,7 @@ func CreateOrUpdateService(ctx context.Context, sclient clientv1.ServiceInterfac
 		}
 	} else {
 		svc.ResourceVersion = service.ResourceVersion
-		svc.Spec.IPFamily = service.Spec.IPFamily
+		svc.Spec.IPFamilies = service.Spec.IPFamilies
```
Comment (lilic) [inline pkg/k8sutil/k8sutil.go:121]: Have not tested this yet, lets see if tests complain, but I suspect it should be as easy as this 🤞

File: go.mod [hunk 2]
```diff
@@ -22,21 +22,21 @@ require (
 	github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring v0.44.1
 	github.com/prometheus-operator/prometheus-operator/pkg/client v0.0.0-00010101000000-000000000000
 	github.com/prometheus/alertmanager v0.21.0
-	github.com/prometheus/client_golang v1.8.0
-	github.com/prometheus/common v0.14.0
+	github.com/prometheus/client_golang v1.9.0
+	github.com/prometheus/common v0.15.0
 	github.com/prometheus/prometheus v1.8.2-0.20201015110737-0a7fdd3b7696
```
Comment (simonpasquier) [inline go.mod:27]: do we want to bump prometheus and thanos versions here too?

**Review by simonpasquier — APPROVED** [review]

File: go.mod (hunk 2 above)
Comment (lilic) [reply go.mod:27]: Can do!

Comment (lilic) [reply go.mod]: Done, any other ones. I didn't go through the minor ones as we have a lot, but if you can think of any other ones happy to bump!

**Review by metalmatze — APPROVED** [review]
Nice!

**Pull request merged**